libs/api/domains/driving-license-book/src/interceptors/studentId.interceptor.ts

```diff
@@ -0,0 +1,37 @@
+import { Observable } from 'rxjs'
+import { map } from 'rxjs/operators'
+import {
+  Injectable,
+  NestInterceptor,
+  ExecutionContext,
+  CallHandler,
+} from '@nestjs/common'
+import { DrivingLicenseBookStudentForTeacher } from '../models/studentsTeacherNationalId.response'
+
+@Injectable()
+export class StudentIdInterceptor implements NestInterceptor {
+  intercept(context: ExecutionContext, next: CallHandler): Observable<unknown> {
+    const prefix = context.getHandler().name
+    const prefixId = (id: string): string => `${prefix}-${id}`
+    return next.handle().pipe(
+      map(
+        (
+          res:
+            | DrivingLicenseBookStudentForTeacher
+            | Array<DrivingLicenseBookStudentForTeacher>,
+        ) => {
+          const isArray = Array.isArray(res)
+          if (isArray) {
+            return res?.map((student) => {
+              student.id = prefixId(student.id)
+
+              return student
+            })
+          }
+          res.id = prefixId(res.id)
```

Comment:
Instead of modifying the object ID, you could add a field called e.g. `handlerName`. Changing the ID of the object can have unforeseen consequences

Reply:
The interceptor is here since there are multiple objects that get the same ID, something unknown is failing on prod and we're hoping this might fix some problems if it's related to caching.